apis/python/src/tiledbsoma/reindexer.cc

```diff
@@ -57,48 +146,19 @@ void load_reindexer(py::module& m) {
                 size_t length = buffer.shape[0];
                 indexer.map_locations(keys.data(), keys.size(), num_threads);
             })
-        .def(
-            "map_locations",
-            [](IntIndexer& indexer,
-               std::vector<int64_t> keys,
-               int num_threads) {
-                indexer.map_locations(keys.data(), keys.size(), num_threads);
-            })
-        // Perform lookup for a large input array of keys and return the looked
-        // up value array (passing ownership from C++ to python)
+        // Perform lookup for a large input array of keys and writes the
+        // looked up values into previously allocated array (works for the
+        // cases in which python and R pre-allocate the array)
         .def(
             "get_indexer",
             [](IntIndexer& indexer, py::array_t<int64_t> lookups) {
-                auto input_buffer = lookups.request();
-                int64_t* input_ptr = static_cast<int64_t*>(input_buffer.ptr);
-                size_t size = input_buffer.shape[0];
-                auto results = py::array_t<int64_t>(size);
-                auto results_buffer = results.request();
-                size_t results_size = results_buffer.shape[0];
-
-                int64_t* results_ptr = static_cast<int64_t*>(
-                    results_buffer.ptr);
-
-                indexer.lookup(input_ptr, results_ptr, size);
-                return results;
+                return get_indexer_general(indexer, lookups);
             })
-        // Perform lookup for a large input array of keys and writes the looked
-        // up values into previously allocated array (works for the cases in
-        // which python and R pre-allocate the array)
-        .def(
-            "get_indexer",
-            [](IntIndexer& indexer,
-               py::array_t<int64_t> lookups,
-               py::array_t<int64_t>& results) {
-                auto input_buffer = lookups.request();
-                int64_t* input_ptr = static_cast<int64_t*>(input_buffer.ptr);
-                size_t size = input_buffer.shape[0];
-
-                auto results_buffer = results.request();
-                int64_t* results_ptr = static_cast<int64_t*>(
-                    results_buffer.ptr);
-                size_t results_size = input_buffer.shape[0];
-                indexer.lookup(input_ptr, input_ptr, size);
-            });
+        // If the input is not arrow (does not have _export_to_c attribute),
+        // it will be handled using a general input method.
+        .def("get_indexer", [](IntIndexer& indexer, py::object py_arrow_array) {
+            return get_indexer_py_arrow(indexer, py_arrow_array);
+        });
```

Comment:
I think these should have distinct names as we discussed Pybind11 will not handle dispatching to the correct arguments. You can also simplify the binding by just passing in the function name.

```
.def("get_indexer_general", get_indexer_general)
.def("get_indexer_py_arrow", get_indexer_py_arrow);
```

Instead of doing the type checking within the Pybind11 function, they should be done in the Python code. And then call the correct `get_indexer_general` or `get_indexer_py_arrow` for that object.

Reply:
This has a huge implication to our entire indexer API! Changing the API is huge change that has ripple effects. I strongly rather not change API specially given the fact that our API is intentionally copying the pandas one.

Reply:
I don't think this should have any effect on the API. You would implement in Python by doing
```
def get_indexer():
    if(is pyarrow object):
        return get_indexer_py_arrow()
    else:
        return get_indexer_general()
```

Reply:
Oh I see the issue now. I think just keep how you have it for now, but we should have the `clib` stuff just internal. In the future, have a `class IntIndex` that holds a member to `clib.IntIndexer`.